src/linear_algebra/finsupp.lean

```diff
@@ -822,6 +822,63 @@ end prod
 
 end finsupp
 
+section fintype
+
+variables {α M : Type*} (R : Type*) [fintype α] [semiring R] [add_comm_monoid M] [module R M]
+variable (v : α → M)
+
+/-- `fintype.total v f` is the linear combination of vectors in `v` with weights in `f`.
+This variant of `finsupp.total` is defined on fintype indexed vectors. -/
+protected def fintype.total : (α → R) →ₗ[R] M :=
+{ to_fun := λ f, ∑ i, f i • v i,
+  map_add' := λ f g, by { simp_rw [← finset.sum_add_distrib, ← add_smul], refl },
+  map_smul' := λ r f, by { simp_rw [finset.smul_sum, smul_smul], refl } }
+
+lemma fintype.total_apply (f) : fintype.total R v f = ∑ i, f i • v i := rfl
+
+@[simp]
+lemma fintype.total_apply_single (i : α) (r : R) :
+  fintype.total R v (pi.single i r) = r • v i :=
+begin
+  simp_rw [fintype.total_apply, pi.single_apply, ite_smul, zero_smul],
+  rw [finset.sum_ite_eq', if_pos (finset.mem_univ _)]
+end
+
+variables (α M)
+
+/-- `fintype.total v f` is linear wrt `v` when `R` is commutative. -/
+def fintype.ltotal (R : Type*) [comm_semiring R] [module R M] : (α → M) →ₗ[R] (α → R) →ₗ[R] M :=
+{ to_fun := fintype.total R,
+  map_add' := λ u v, by { ext, simp },
+  map_smul' := λ r v, by { ext, simp } }
+
+variables {α M}
+
+lemma finsupp.total_eq_fintype_total_apply (x : α → R) :
```

Comment:
I'd like to make this a `simp` lemma, but the `S` that `fintype.total` is linear in becomes a free variable here. Maybe make a copy for the commutative case where `S = R`? @eric-wieser, do you have an opinion?

Reply:
If we want this, I suggest we add it later.